File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] answer = new int[2];
        for (int i=0, temp=0; i<nums.length; i++) {
            temp = target - nums[i];
            if (map.containsKey(temp)) {
                answer[0] = i;
                answer[1] = map.get(temp);
                break;
            }
            map.put(nums[i], i);
        }
        return answer;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        
        for i in range(len(nums)):
            current_num = nums[i]
            complement = target - current_num
            
            if complement in dic:
                return [dic[complement], i]
            
            dic[current_num] = i
        
        return []